docker/docker-compose.yml

```diff
@@ -0,0 +1,141 @@
+# Auto generates the properties files for all services and the nginx config
+configs:
+  image: python:2.7
+  volumes:
+    - "./entrypoints:/entrypoints"
+    - "./configs/generators:/configs/generators"
+    - "$HOME/.aws/:/root/.aws:ro"
+    - "$HOME/.gu/grid/:/root/.gu/grid"
+    - "/configs"
+  entrypoint: /entrypoints/configs.sh
+
+# Proxy server for all services
+nginx:
+  image: nginx:latest
+  links:
+    - elasticsearch
+    - media-api
+    - thrall
+    - image-loader
+    - kahuna
+    - cropper
+    - metadata-editor
+    - usage
+    - collections
+    - imgops
+  ports:
+    - "80:80"
+    - "443:443"
+  volumes:
+    - "./entrypoints:/entrypoints"
+  volumes_from:
+    - configs
+  entrypoint: /entrypoints/nginx.sh
+
+# Shared volume container
+data:
+  image: java:8
+  volumes:
+    - "../:/code"
+    - "/root/.ivy2"
+    - "/root/.sbt"
+
+elasticsearch:
+  build: .
+  dockerfile: Dockerfile-elastic
+  container_name: elasticsearch
+  expose:
+    - "9200"
+    - "9300"
+  ports:
+    - "9200:9200"
+    - "9300:9300"
+  volumes:
+    - "./configs/elasticsearch:/usr/share/elasticsearch/config"
```

Comment:
Might also be a good idea to mount a host directory to the `/usr/share/elasticsearch/data` volume so that the data is persisted outside of the container.